lib/cli/interactive-setup/console-instrument.js

```diff
@@ -0,0 +1,202 @@
+
+  return {
+    isInstrumented: total === context.targetFunctions.length,
+    total: context.targetFunctions.length,
+    instrumented: total,
+  };
+};
+
+const waitForInstrumentation = async (context) => {
+  const instrumentationProgress = progress.get(progressKey);
+  let isInstrumenting = true;
+  while (isInstrumenting) {
+    const { isInstrumented: done, total, instrumented } = await checkInstrumentationStatus(context);
+    instrumentationProgress.update(`Instrumenting ${instrumented}/${total} functions`);
+    if (done) {
+      isInstrumenting = false;
+    } else {
+      await wait(1000);
+    }
+  }
+};
+
+module.exports = {
+  async isApplicable(context) {
+    const { isConsole, enableDev } = context;
+
+    if (!isConsole) {
+      context.inapplicabilityReasonCode = 'NON_CONSOLE_CONTEXT';
+      return false;
+    }
+
+    if (!enableDev) {
+      context.inapplicabilityReasonCode = 'NO_ENABLE_DEV';
+      return false;
+    }
+
+    if (!context.serverless) {
+      context.inapplicabilityReasonCode = 'NO_SERVERLESS_CONTEXT';
+      return false;
+    }
+
+    if (!(await resolveAuthMode())) {
+      context.inapplicabilityReasonCode = 'NOT_LOGGED_IN';
+      return false;
+    }
+
+    if (!context.org) {
+      context.inapplicabilityReasonCode = 'UNRESOLVED_ORG';
+      return false;
+    }
+
+    // Add single function name or all function names to the list
+    const functionNames = [];
+    const osFunctions = [];
+    if (context.options.function) {
+      const functionName = `${context.serverless.service.service}-${context.serverless.service.provider.stage}-${context.options.function}`;
+      osFunctions.push({
+        instrumentations: {
+          mode: 'dev',
+        },
+        resourceKey: `aws_${context.awsAccountId}_function_${context.serverless.service.provider.region}_${functionName}`,
+      });
+      functionNames.push(functionName);
+    } else {
+      const names = Object.keys(context.serverless.service.functions);
+      for (const name of names) {
+        const functionName = `${context.serverless.service.service}-${context.serverless.service.provider.stage}-${name}`;
+        // console.log(functionName);
+        osFunctions.push({
+          instrumentations: {
+            mode: 'dev',
+          },
+          resourceKey: `aws_${context.awsAccountId}_function_${context.serverless.service.provider.region}_${functionName}`,
+        });
+        functionNames.push(functionName);
+      }
+    }
+
+    context.targetInstrumentations = osFunctions;
+    context.targetFunctions = functionNames;
+
+    const { allExist, total, functionCount } = await allFunctionsExist(context);
+    if (!allExist) {
+      log.notice(
+        `Only ${functionCount} of ${total} functions exist on this integration. Be sure to deploy the service before using dev.`
+      );
+      context.deployNeeded = true;
+      return false;
+    }
+
+    const { isInstrumented } = await checkInstrumentationStatus(context);
```

Comment:
Yeah it would fail. When we wait for all functions to be instrumented we need be sure that all of the functions we requested are attached to our integration. If they are not then the waiting for instrumentation will stall indefinitely.